### ISSUE TITLE
How i can use russian language?..

### ISSUE THREAD

**Issue by musaev-haybulla** [Respect/Validation#645]
In this library function of validate alnum, alpha and other cannot support russian language. How can fix it?..

**Comment by henriquemoody** [comment]
It depends on the locale that is set. Try to define it in your bootstrap:

```php
setlocale(LC_ALL, 'ru_RUS.utf8');
```

I'm gonna wait for you feedback

**Comment by musaev-haybulla** [comment]
I did as You wrote. Function ctype_alnum, like other ctype functions, not vosprinimaut changing the system locale to Cyrillic in utf-8. Neither on my server nor on a commercial server, which I configured not. Any thoughts on this?

**Comment by Savsan** [comment]
Hi! I've a same problem. Russians characters can't be validate, even after i changed locale on 1251, UTF-8. How can fix it?

**Comment by musaev-haybulla** [comment]
I solved this problem without making code changes. But this is not correct, because if I forget and load the composer om new version of the library, as immediately erased all what I wrote in the code. Knowledgeable guys suggested to make a fork of this library and to scratch code. Namely, to replace all the ctype functions (the benefit there is not a lot) on regular expressions.

on Russiain
Я не решил эту проблему, не внося изменений в код. А это не правильно, так как стоит мне забыться и подгрузить composer'ом новые версии библиотеки, как тут же сотрется все то, что я писал в коде. Знающие ребята подсказали сделать fork от этой библиотеки и поковырять код. А именно - заменить все ctype функции (благо их там не много), на регулярные выражения.

**Comment by daffoxdev** [comment]
I have created rule anyLetter that check for regexp for any utf-8 letter. You can create same only for russian. Read how to create own rules for this library.

```php
namespace YourProjectNamespace\Validation\Rules;

use Respect\Validation\Rules\AbstractRule;

class AnyLetter extends AbstractRule
{
    public function validate($input)
    {
        if (preg_match('/^\pL+$/u', $input)) {
            return true;
        } else {
            return false;
        }
    }
}
```


and Exception
```php
namespace YourProjectNamespace\Validation\Exceptions;

use Respect\Validation\Exceptions\ValidationException;

class AnyLetterException extends ValidationException
{
    public static $defaultTemplates = [
        self::MODE_DEFAULT => [
            self::STANDARD => '{{name}} must contain only letters',
        ],
        self::MODE_NEGATIVE => [
            self::STANDARD => '{{name}} must not contain only letters',
        ],
    ];
}
```

Something like this. I think you got the main idea. Its clear way to not changing source code as you did.
 Just take idea and write own rules and use them as you wish.

PS. Need to do class for test, but I haven't done it still :)

**Comment by Savsan** [comment]
Thanks!

**Comment by henriquemoody** [comment]
I created #762 in order to fix this issue.

However, before I merge it I would like to understand in which scenario do you intent to use this validator?

**Comment by Savsan** [comment]
Hello! I make my first, simple app and i used validator for form in controllers of slim 3 microframework. Sorry for my English, I am studying it yet. Have you already written a regular expression for Russian language?

**Comment by Savsan** [comment]
Oh, sorry for my question. I didn't see that you have already written rurels.

**Comment by henriquemoody** [comment]
No problem, but still, what the validation you want to perform? Which field, for example?

**Comment by Savsan** [comment]
For example: names of places, cities, rivers, lakes, seas, oceans, first - last - middle names of people (can contain an apostrophe, if it'll be transliteration from foreign languages), addresses, education and .... )

Russian text (text areas for example) can contain a signs like;  ' " : ; !  ( ) , + - * / = . and also english symbols

Email addresses, in Russia - only on English.
Login areas like rule on English.
Password maybe on Russian.

DataTime: (comparison RUS - USA)
RUS: DD.MM.YYYY, 24.01.2011
USA: MM-DD-YYYY, 01-24-2011

**Comment by musaev-haybulla** [comment]
henriquemoody, 
this opportunity I need to check some of the form fields (name, surname, and so on.) in the absence of any unnecessary characters.

You could not say how soon your pull request with these long-awaited revisions will be commit? ..

**Comment by henriquemoody** [comment]
Now is more clear what is the intention here.

According to [Wikipedia](https://en.wikipedia.org/wiki/Alphanumeric):
> Alphanumeric is a combination of alphabetic and numeric characters, and is used to describe the collection of Latin letters and Arabic digits or a text constructed from this collection.

That said, the one we have is correct, and changing it would make it wrong. What is needed is a new rule, with a new name and a new purpose. I'm planning to finish it this week, and I'm gonna make it more clear on the documentation. The rule will be probably called _Name_ but I'm very open for suggestions.

**Comment by nickl-** [comment]
Closed: as per #762 Won't fix
